src/main/java/com/targus/experiment/Experiment.java

```diff
@@ -27,6 +28,57 @@ public static void writeToFile(String fileName, String text, boolean append) {
 
     }
 
+    // TODO: reduce the number of parameters this method takes
+    // TODO: check the if condition in detail
+    public static void writeToJson(WSN wsn, Point2D dimensions, int terminationValue, int communicationRange, int sensingRange, String fileName, boolean override) {
+        String filePath = Constants.DEFAULT_BASE_PATH_FOR_JSON_FILES + fileName;
+        File f = new File(filePath);
+        if(f.exists() && !f.isDirectory() && !override) {
+            System.out.println(fileName + " already exists in " + Constants.DEFAULT_BASE_PATH_FOR_JSON_FILES + ". No changes made to the file.");
+            System.out.println("\nIf you want to make changes to the file, you may provide 'true' for the override parameter\n");
+            return;
+        }
+
+        BufferedWriter writer;
+        try {
+            writer = Files.newBufferedWriter(Paths.get(filePath));
+            ObjectMapper objectMapper = new ObjectMapper();
+
+            Map<String, Object> problemInfo = new HashMap<>();
+            problemInfo.put(Constants.DIMENSIONS, Arrays.asList(dimensions.getX(), dimensions.getY()));
+
+            List<double[]> targetList = new ArrayList<>();
+            for (Point2D target : wsn.getTargets()) {
+                double[] coords = new double[2];
+                coords[0] = target.getX();
+                coords[1] = target.getY();
+                targetList.add(coords);
+            }
+            problemInfo.put(Constants.TARGETS, targetList);
+
+            List<double[]> potentialPositionList = new ArrayList<>();
+            for (Point2D potentialPosition : wsn.getPotentialPositions()) {
+                double[] coords = new double[2];
+                coords[0] = potentialPosition.getX();
+                coords[1] = potentialPosition.getY();
+                potentialPositionList.add(coords);
+            }
+            problemInfo.put(Constants.POTENTIAL_POSITIONS, potentialPositionList);
+
+            problemInfo.put(Constants.COMMUNICATION_RADIUS, communicationRange);
+            problemInfo.put(Constants.SENSING_RADIUS, sensingRange);
+            problemInfo.put(Constants.M, wsn.getM());
+            problemInfo.put(Constants.K, wsn.getK());
+            problemInfo.put(Constants.GENERATION_COUNT, terminationValue);
+            problemInfo.put(Constants.MUTATION_RATE, wsn.getMutationRate());
+
+            writer.write(objectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(problemInfo));
+            writer.close();
+        } catch (IOException e) {
+            throw new RuntimeException("Something went wrong when writing to json file. " + e);
+        }
+    }
+
```

Comment:
Refactoring will be done for a couple of methods and one of those methods is exportToFile method. When it is done, I will recheck this portion of code to see if I can replace this method here. I will let you know.